### PR TITLE
Add logout endpoint

### PR DESCRIPTION
**Pull request by sarataha**
Closes: https://github.com/weaveworks/weave-gitops-enterprise/issues/407

**What changed?**
Added a new /logout endpoint to be called by FE for cookie deletion.

**Why?**
Part of Auth implementation.

**How did you test it?**
Tested locally + unit tests.

**Release notes**

**Documentation Changes**

### CODE REVIEW

**Review by yiannistri — APPROVED** [review]
Nice one! Can we add a couple of unit tests for failure/success too? 🙏

File: pkg/server/auth/server_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/oauth2-proxy/mockoidc"
+	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+	"gotest.tools/v3/assert"
```
Comment (foot) [inline pkg/server/auth/server_test.go:14]: Lets use the normal testify lib here

File: pkg/server/auth/server_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		Name:     "id_token",
+		Value:    "foo",
+		Path:     "/",
+		Expires:  time.Now().UTC().Add(5),
+		HttpOnly: true,
+	}
+
+	w := httptest.NewRecorder()
+
+	http.SetCookie(w, cookie)
```
Comment (foot) [inline pkg/server/auth/server_test.go:44]: This line is introducing the 2nd cookie in the response, we can remove it 👍

**Review by foot — APPROVED** [review]
💯

**Pull request merged**